eclipse-theia/theia
# display tree node loading on slow connections

#### What it does


- fix #3450
  - mark node as busy after 800ms during refreshing
  - new api `TreeModel.markAsBusy(node, ms, cancellationToken)`  to allow clients indicate that some task is performed for the given node (makes visul effect currently only for composite nodes)
- fix #5959
  - report navigator busy node progress in `explorer` view container
  - new progress location `explorer` for the view container

Copied icons were already were approved in previous CQs.

![loading](https://user-images.githubusercontent.com/3082655/75623828-8d394800-5bae-11ea-8c40-a6462dcdeb9c.gif)

#### How to test


- emulate slow connection on the system level, for instance with `Network Line Conditioner` or Charles2 proxy
- try to expand some nodes in different trees
  - you should see loading indicator in any tree (i.e. try in npm scripts for the plugin system)
  - you should see blueish bar in the explorer view container for the navigator tree

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
